sockeye/translate.py

```diff
@@ -55,52 +52,95 @@ def main():
             restrict_lexicon.load(args.restrict_lexicon)
-        translator = sockeye.inference.Translator(context,
-                                                  args.ensemble_mode,
-                                                  args.bucket_width,
-                                                  sockeye.inference.LengthPenalty(args.length_penalty_alpha,
-                                                                                  args.length_penalty_beta),
-                                                  models,
-                                                  vocab_source,
-                                                  vocab_target,
-                                                  restrict_lexicon)
-        read_and_translate(translator, output_handler, args.chunk_size, args.input)
-
-
-def read_and_translate(translator: sockeye.inference.Translator, output_handler: sockeye.output_handler.OutputHandler,
-                       chunk_size: Optional[int], source: Optional[str] = None) -> None:
+        translator = inference.Translator(context=context,
+                                          ensemble_mode=args.ensemble_mode,
+                                          bucket_source_width=args.bucket_width,
+                                          length_penalty=inference.LengthPenalty(args.length_penalty_alpha,
+                                                                                 args.length_penalty_beta),
+                                          models=models,
+                                          source_vocabs=source_vocabs,
+                                          target_vocab=target_vocab,
+                                          restrict_lexicon=restrict_lexicon)
+        read_and_translate(translator=translator,
+                           output_handler=output_handler,
+                           chunk_size=args.chunk_size,
+                           input=args.input,
+                           input_factors=args.input_factors,
+                           json_input=args.json_input)
+
+
+def make_inputs(input: Optional[str],
+                translator: inference.Translator,
+                json_input: bool,
+                input_factors: Optional[List[str]] = None) -> Generator[inference.TranslatorInput, None, None]:
+    """
+    Generates TranslatorInput instances from input. If input is None, reads from stdin. If num_input_factors > 1,
+    the function will look for factors attached to each token, separated by '|'.
+    If source is not None, reads from the source file. If num_source_factors > 1, num_source_factors source factor
+    filenames are required.
+
+    :param input: The source file (possibly None).
+    :param translator: Translator that will translate each line of input.
+    :param json_input: Whether the input is in json format.
+    :param input_factors: Source factor files.
+    :return: TranslatorInput objects.
+    """
+    if input is None:
```

Comment:
might make sense to name `input` differently as `input` is a builtin.

Reply:
renamed to inp (and inp_factors)